### ISSUE TITLE
ghost tables: error in `CDB_SaveDDLTransaction()`

### ISSUE THREAD

**Issue by owayss** [CartoDB/cartodb-postgresql#368]
**version**: `0.28.1`

I suspect that due to the limited `search_path` during pg_restore, Postgres is unable to find the constraint `cdb_ddl_execution_pkey` constaint:
```
ERROR:  constraint "cdb_ddl_execution_pkey" for table "cdb_ddl_execution" does not exist
```
This is an issue when the `cdb_ddl_execution` trigger is being restored:
```
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 txid   | integer |           | not null | 
 tag    | text    |           |          | 
Indexes:
    "cdb_ddl_execution_pkey" PRIMARY KEY, btree (txid)
Triggers:
    check_ddl_update AFTER INSERT ON cdb_ddl_execution DEFERRABLE INITIALLY DEFERRED FOR EACH ROW EXECUTE PROCEDURE _cdb_linkghosttablestrigger()
```
As is ultimately calls the `CDB_SaveDDLTransaction()` func, which contains the -now- problematic upsert statement: https://github.com/CartoDB/cartodb-postgresql/blob/master/scripts-available/CDB_GhostTables.sql

**Comment by Algunenano** [comment]
The issue is that `txid` isn't a primary key in that table, so `cdb_ddl_execution_pkey` doesn't exists. I don't know how that happened, so it'll need to be investigated.

Currently it's created as primary:
https://github.com/CartoDB/cartodb-postgresql/blob/4c93258cd2d93e530d474d7f7b124027cd973400/scripts-available/CDB_GhostTables.sql#L99

But we should check if this has always been the case.

**Comment by oleurud** [comment]
Showing the function code:
```
    INSERT INTO @extschema@.cdb_ddl_execution VALUES (txid_current(), tg_tag) ON CONFLICT (txid) DO NOTHING;
```
it seems to be correct and I agree the problem looks like the `search_path` does not have `cartodb`.

**Comment by gonzaloriestra** [comment]
No, I think we could disable it in the destination database before starting, and then enable it again when it finishes. And finally run the process to sync everything.

- To disable: `SELECT cartodb.CDB_DisableGhostTablesTrigger()` or `user.db_service.drop_ghost_tables_event_trigger`
- To enable: `SELECT cartodb.CDB_EnableGhostTablesTrigger()` or `user.db_service.create_ghost_tables_event_trigger`
- To run ghost tables: `Carto::GhostTablesManager.new(user.id).link_ghost_tables_synchronously`

**Comment by alrocar** [comment]
We merged a fix. I'm closing this one.
